pythainlp/tokenize/newmm.py

```diff
@@ -110,4 +115,52 @@ def segment(text: str, custom_dict: Trie = None) -> List[str]:
     if not custom_dict:
         custom_dict = DEFAULT_DICT_TRIE
 
-    return list(_onecut(text, custom_dict))
+    text_len = len(text)
+
+    # if the text is longer than the limit,
+    # breaks them into smaller chunks then tokenizes each chunk
+    if text_len >= (_TEXT_LIMIT + _TEXT_SCAN_RIGHT):
+        text_parts = []
+
+        while text_len >= (_TEXT_LIMIT + _TEXT_SCAN_RIGHT):
+            sample_start = _TEXT_LIMIT - _TEXT_SCAN_LEFT
+            sample_end = _TEXT_LIMIT + _TEXT_SCAN_RIGHT
+            sample = text[sample_start:sample_end]
+
+            # find possible break positions
+            cut_pos = sample_end
+
+            # try to break by space first
+            space_idx = sample.rfind(" ")
```

Comment:
what happens if the space is in the middle of the name, e.g. ` ....ชื่อ นามสกุล.....`? Am I right that the algorithm probably tokenises wrongly in this case?

Reply:
Yes. I will unfortunately breaks at the space.
This will also include case like "ปอน ๆ"